From 5137c82720cdb66f8e97e53ccd8a28f082cc3bda Mon Sep 17 00:00:00 2001
From: Lubosz Sarnecki <lubosz@gmail.com>
Date: Mon, 17 Nov 2025 16:50:28 +0100
Subject: [PATCH 2/2] rocroller: Always fetch libdivide if it's not installed.

When not found, fetch libdivide, even if fetching deps is turned off.
This can patch can be dropped if libdivie is packaged.
---
 shared/rocroller/CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/shared/rocroller/CMakeLists.txt b/shared/rocroller/CMakeLists.txt
index 52bfdd0f89..85b4556d2d 100644
--- a/shared/rocroller/CMakeLists.txt
+++ b/shared/rocroller/CMakeLists.txt
@@ -106,7 +106,6 @@ endif()
 
 find_package(libdivide QUIET)
 if(NOT libdivide_FOUND)
-if(ROCROLLER_ENABLE_FETCH)
     FetchContent_Declare(
         libdivide
         GIT_REPOSITORY https://github.com/ridiculousfish/libdivide.git
@@ -114,9 +113,6 @@ if(ROCROLLER_ENABLE_FETCH)
     )
     FetchContent_MakeAvailable(libdivide)
     set_property(DIRECTORY ${libdivide_SOURCE_DIR} PROPERTY EXCLUDE_FROM_ALL ON)
-    else()
-        message(FATAL_ERROR "Failed to find libdivide")
-    endif()
 endif()
 
 find_package(msgpack-cxx CONFIG)
-- 
2.51.2

